support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -142,13 +151,57 @@ class CreateSubscriptionController(
     }
   }
 
+  private def validateBenefits(response: UserBenefitsResponse): EitherT[Future, CreateSubscriptionError, Unit] = {
+    if (response.benefits.contains("adFree")) {
+      // Not eligible
+      EitherT.leftT(RequestValidationError("guardian_light_purchase_not_allowed"))
+    } else {
+      // Eligible
+      EitherT.rightT(())
+    }
+  }
+
+  private def validateUserIsEligibleForPurchase(
+      request: Request[CreateSupportWorkersRequest],
+      userDetails: UserDetailsWithSignedInStatus,
+  ): EitherT[Future, CreateSubscriptionError, Unit] = {
+    request.body.product match {
+      case GuardianLight(_) => {
+        if (userDetails.isSignedIn) {
+          // If the user is signed in, we'll assume they're eligible as they shouldn't have got
+          // to this point of the journey.
+          EitherT.rightT(())
+        } else {
+          for {
+            benefits <- userBenefitsApiService
+              .getUserBenefits(userDetails.userDetails.identityId)
+              .leftMap(_ => ServerError("Something went wrong calling the user benefits API"))
```

Comment:
If there's an issue with calling the user benefits API the purchase will be blocked and we'll show a generic error message to the user.